processing/src/main/java/io/druid/query/filter/ValueMatcherFactory.java

```diff
@@ -28,5 +28,6 @@
 {
   public ValueMatcher makeValueMatcher(String dimension, String value);
   public ValueMatcher makeValueMatcher(String dimension, Predicate<String> value);
+  public ValueMatcher makeValueMatcher(String[] dimension, Predicate<Object[]> value, boolean cartesian);
```

Comment:
lets add javadocs to explain what cartesian means. It might also be clearer to have two methods rather than a flag and have the method name explain the difference.